dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/persistence/file/FileExtTest.kt

```diff
@@ -0,0 +1,525 @@
+    }
+
+    @Test
+    fun `M return null W readTextSafe() {file doesn't exist}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn false
+
+        // When
+        val readText = mockFile.readTextSafe()
+
+        // Then
+        assertThat(readText).isNull()
+    }
+
+    @Test
+    fun `M return null W readTextSafe() {file can't be read}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn true
+        whenever(mockFile.canRead()) doReturn false
+
+        // When
+        val readText = mockFile.readTextSafe()
+
+        // Then
+        assertThat(readText).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readTextSafe() {SecurityException}`(
+        @StringForgery message: String
+    ) {
+        // Given
+        whenever(mockFile.name) doThrow SecurityException(message)
+
+        // When
+        val readText = mockFile.readTextSafe()
+
+        // Then
+        assertThat(readText).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readTextSafe() {FileNotFoundException}`() {
+        // When
+        val readText = fakeFile.readTextSafe()
+
+        // Then
+        assertThat(readText).isNull()
+    }
+
+    @Test
+    fun `M return result W readLinesSafe()`(
+        @StringForgery result: List<String>
+    ) {
+        // Given
+        fakeFile.writeText(result.joinToString("\n"))
+
+        // When
+        val readLines = fakeFile.readLinesSafe()
+
+        // Then
+        assertThat(readLines).isEqualTo(result)
+    }
+
+    @Test
+    fun `M return result W readLinesSafe() {custom charset}`(
+        @StringForgery result: List<String>,
+        @Forgery charset: Charset
+    ) {
+        // Given
+        fakeFile.writeText(result.joinToString("\n"), charset)
+
+        // When
+        val readLines = fakeFile.readLinesSafe(charset)
+
+        // Then
+        assertThat(readLines).isEqualTo(result)
+    }
+
+    @Test
+    fun `M return null W readLinesSafe() {file doesn't exist}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn false
+
+        // When
+        val readLines = mockFile.readLinesSafe()
+
+        // Then
+        assertThat(readLines).isNull()
+    }
+
+    @Test
+    fun `M return null W readLinesSafe() {file can't be read}`() {
+        // Given
+        whenever(mockFile.exists()) doReturn true
+        whenever(mockFile.canRead()) doReturn false
+
+        // When
+        val readLines = mockFile.readLinesSafe()
+
+        // Then
+        assertThat(readLines).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readLinesSafe() {SecurityException}`(
+        @StringForgery message: String
+    ) {
+        // Given
+        whenever(mockFile.name) doThrow SecurityException(message)
+
+        // When
+        val readLines = mockFile.readLinesSafe()
+
+        // Then
+        assertThat(readLines).isNull()
+    }
+
+    @Test
+    fun `M catch exception W readLinesSafe() {FileNotFoundException}`() {
+        // When
```

Comment:
should we mock behavior with `FileNotFoundException` here?

Reply:
The fakeFile by definition doesn't exist, so we're guaranteed to trigger a FNFE

Reply:
It seems it won't be triggered, because in the `readLinesSafe` call `existsSafe` will return `false`, so it won't come to the point when `readLines` is executed (and I guess we want to test FNFE exception thrown by this call).